src/Autofac/Features/Scanning/TypeExtensions.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Autofac Project. All rights reserved.
```

Comment:
This all came out of the OpenGenericScanningRegistrationExtensions and ScanningRegistrationExtensions classes.